ubuntu_server_triage.py

```diff
@@ -71,37 +80,47 @@ def bug_info(bugs):
     takes a considerable amount of time.
     """
     bug_list = []
-    for bug in bugs:
+    for (bug, status) in bugs:
         num = bug.split(' ')[1].replace('#', '')
         src = bug.split(' ')[3]
         title = ' '.join(bug.split(' ')[5:]).replace('"', '')
-        bug_list.append((num, src, title))
+        bug_list.append((num, src, title, status))
 
     bug_list.sort(key=lambda tup: tup[0])
 
     return bug_list
 
 
-def modified_bugs(date):
+def modified_bugs(date, lpname, bugsubscriber):
     """
     Returns a list of bugs modified after a specific date.
     """
     # Distribution List: https://launchpad.net/distros
     # API Doc: https://launchpad.net/+apidoc/1.0.html
     launchpad = connect_launchpad()
     project = launchpad.distributions['Ubuntu']
-    team = launchpad.people['ubuntu-server']
-
-    raw_bugs = project.searchTasks(modified_since=date,
-                                   structural_subscriber=team)
-
-    bugs = [bug.title for bug in raw_bugs]
+    team = launchpad.people[lpname]
+
+    if bugsubscriber:
+        # direct subscriber
+        raw_bugs = project.searchTasks(modified_since=date,
+                                       bug_subscriber=team)
+    else:
+        # structural_subscriber sans already subscribed
+        mod_bugs = project.searchTasks(modified_since=date,
+                                       structural_subscriber=team)
+        already_sub_bugs = project.searchTasks(modified_since=date,
```

Comment:
Execution time isn't so bad IMO, not as long as we don't start to scale with the list length too much.
But I see that you might want to see those we are already subscribed, now that we have the status added it might be less annoying to have those already subscribed in the list.

I'd suggest we make this an option which is off-by-default.
But actually while I was writing this I think it goes way better.
Avoid the clumsy extra option and just mark bugs that we are already subscribed in the overview.
I did that and the view is now simple:
not subscribed
(In Progress)
subscribed
*(In Progress)